design/src/docs/anatomy.md

```diff
@@ -0,0 +1,755 @@
+
+During the survey we will use the [`pokemon.smithy`](https://github.com/awslabs/smithy-rs/blob/main/codegen-core/common-test-models/pokemon.smithy) model as a reference:
+
+```smithy
+/// The Pokémon Service allows you to retrieve information about Pokémon species.
+@title("Pokémon Service")
+@restJson1
+service PokemonService {
+    version: "2021-12-01",
+    resources: [PokemonSpecies, Storage],
+    operations: [
+        GetServerStatistics,
+        DoNothing,
+        CapturePokemon,
+        CheckHealth
+    ],
+}
+```
+
+Smithy Rust will use this model to produce the following API:
+
+```rust
+// A handler for the `GetPokemonSpecies` operation (the `PokemonSpecies` resource).
+async fn get_pokemon_species(input: GetPokemonSpeciesInput) -> Result<GetPokemonSpeciesOutput, GetPokemonSpeciesError> {
+    /* implementation */
+}
+
+// Apply a `tower::Layer` to a handler.
+let get_pokemon_species_op = GetPokemonSpecies::from_handler(get_pokemon_species).layer(/* some `tower::Layer` */);
+
+// Use the service builder to create `PokemonService`.
+let pokemon_service = PokemonService::builder()
+    // Pass the handler directly to the service builder...
+    .get_pokemon_species(get_pokemon_species)
+    // ...or pass the layered handler.
+    .get_pokemon_species_operation(get_pokemon_species_op)
+    /* other operation setters */
+    .build();
+```
+
+## Operations
+
+A [Smithy Operation](https://awslabs.github.io/smithy/2.0/spec/service-types.html#operation) specifies the input, output, and possible errors of an API operation. One might characterize a Smithy Operation as syntax for specifying a function type.
+
+We represent this in Rust using the [`OperationShape`](https://github.com/awslabs/smithy-rs/blob/4c5cbc39384f0d949d7693eb87b5853fe72629cd/rust-runtime/aws-smithy-http-server/src/operation/shape.rs#L8-L22) trait:
+
+```rust
+pub trait OperationShape {
+    /// The name of the operation.
+    const NAME: &'static str;
+
+    /// The operation input.
+    type Input;
+    /// The operation output.
+    type Output;
+    /// The operation error. [`Infallible`](std::convert::Infallible) in the case where no error
+    /// exists.
+    type Error;
+}
+```
+
+For each Smithy Operation shape,
+
+```smithy
+/// Retrieve information about a Pokémon species.
+@readonly
+@http(uri: "/pokemon-species/{name}", method: "GET")
+operation GetPokemonSpecies {
+    input: GetPokemonSpeciesInput,
+    output: GetPokemonSpeciesOutput,
+    errors: [ResourceNotFoundException],
+}
+```
+
+the following implementation is generated
+
+```rust
+/// Retrieve information about a Pokémon species.
+pub struct GetPokemonSpecies;
+
+impl OperationShape for GetPokemonSpecies {
+    const NAME: &'static str = "com.aws.example#GetPokemonSpecies";
+
+    type Input = GetPokemonSpeciesInput;
+    type Output = GetPokemonSpeciesOutput;
+    type Error = GetPokemonSpeciesError;
```

Comment:
This type, `GetPokemonSpeciesError`, doesn't appear in the smithy file, I assume it's a generated (rust) enum of the possible errors, but maybe it could use an introduction?